v3/src/components/case-table/inspector-panel/trash-menu-list.tsx

```diff
@@ -0,0 +1,75 @@
+import { MenuItem, MenuList } from "@chakra-ui/react"
+import React from "react"
+import { useDataSetContext } from "../../../hooks/use-data-set-context"
+import { IAttribute } from "../../../models/data/attribute"
+import t from "../../../utilities/translation/translate"
+
+export const TrashMenuList = () => {
+  const data = useDataSetContext()
+
+  const createGhostCase = (attrs:IAttribute[]) => {
+    const gCase:Record<string, any> = {}
+    attrs?.forEach(attr => {
+      gCase[attr.name]=""
+    })
+    return gCase
+  }
+
+  const handleSelectAllCases = () => {
+    const caseIDArr:string[] = []
+    if (data) {
+      data.cases.forEach(c => {
+        caseIDArr.push(c.__id__)
+      })
+      data.selectCases(caseIDArr)
+    }
+  }
+
+  const handleDeleteSelectedCases = () => {
+    if (data) {
+      data.removeCases(Array.from(data.selection))
+      if (data.cases.length === 0) {
+        const ghostCase = createGhostCase(data.attributes)
+        data.addCases([ghostCase])
+      }
+    }
+  }
+
+   const handleDeleteUnselectedCases = () => {
+    const unselectedCaseIDArr:string[] = []
+
+    if (data) {
+      const casesToRemove = data.cases.filter(c=>
+        !((Array.from(data.selection)).includes(c.__id__)))
+      casesToRemove.forEach(c => {
+        unselectedCaseIDArr.push(c.__id__)
+      })
+      data.removeCases(unselectedCaseIDArr)
```

Comment:
```suggestion
      const casesToRemove = data.cases.filter(c => !data.selection.has(c.__id__)).map(c => c.__id__);
      data.removeCases(casesToRemove)
```
Converting to `Array` is inefficient and `Array.includes` is less efficient than `Set.has`.